src/java.base/windows/native/libnio/ch/Net.c

```diff
@@ -554,7 +550,11 @@ Java_sun_nio_ch_Net_blockOrUnblock6(JNIEnv *env, jobject this, jboolean block, j
     int opt = (block) ? MCAST_BLOCK_SOURCE : MCAST_UNBLOCK_SOURCE;
     int n = setGroupSourceReqOption(env, fdo, opt, group, index, source);
     if (n == SOCKET_ERROR) {
-        handleSocketError(env, WSAGetLastError());
+        if (block) {
+            NET_ThrowNew(env, WSAGetLastError(), "setsocketopt to block source");
+        } else {
+            NET_ThrowNew(env, WSAGetLastError(), "setsocketopt to unblock source");
```

Comment:
I think this can be reduced to one message (block or unblock) source as it doesn't add value as the stack trace will make it very clear if a source-specific block or unblock op.

Reply:
Okay, I reduced it to one message.